src/v/cluster/tm_stm.h

```diff
@@ -252,6 +254,96 @@ class tm_stm final : public persisted_stm {
     }
 };
 
+struct tm_transaction_v1 {
```

Comment:
Yes.
I did it the same like Denis did for v1.
So it can add problems with serialization. But maybe we have another way? @rystsov ?

Reply:
We're working with complex logic here and we want to ease cognitive load caused by accidental stuff and let a reader to focus on the essential logic first. Currently the transactional coordinator works only with the latest version and the v1, v0 to the latest conversion is contained in a small area of the code so the reader should keep in mind interaction with the latest version only.

If we go for 

```
struct tm_transaction_v1 {
tm_transaction_v0 v0;
}
```

Then the reader will be exposed to the previous versions too. The optimization of the copy code doesn't worth the increased cognitive load. Yes it's possible to forget to copy something but we have excessive chaos tests which help us to catch it as soon as it happens.

Another benefit is the ability to drop v0 nad v1 without rewriting the coordinator logic. The decision introduce _v0 and _v1 isn't dictated by the adl limitations. It's a concision decision to avoid coupling and keep the essential code simple.